src/NHSD.BuyingCatalogue.Solutions.Application/Commands/ClientApplications/NativeMobile/UpdateSolutionNativeMobileFirst/UpdateSolutionNativeMobileFirstHandler.cs

```diff
@@ -22,7 +22,7 @@ public UpdateSolutionNativeMobileFirstHandler(ClientApplicationPartialUpdater cl
         public async Task<ISimpleResult> Handle(UpdateSolutionNativeMobileFirstCommand request,
             CancellationToken cancellationToken)
         {
-            var validationResult = _updateSolutionNativeMobileFirstValidator.Validation(request);
+            var validationResult = UpdateSolutionNativeMobileFirstValidator.Validation(request);
```

Comment:
Please review this line. The `UpdateSolutionNativeMobileFirstValidator` instance was previously passed in as a dependency, which I assume would allow easier testing. Does this have to be made static? If so could the dependency be removed? How will this effect the tests?

Reply:
Sadly it does have to made static, so the dependency is removed. The tests all pass

Reply:
I don't think this does need to be made `static`.

Reply:
Sorry, just commenting as I'd only just noticed you'd picked this up. I was planning to pick this up as I already a stash with the package updates.

I've just checked my stashed changes and I did need to make a change here but it looks like I didn't make it `static`.

I think I noticed there was a flaw in the implementation and fixed that instead.

Reply:
Would you be able to tell me the flaw? @bjss-chris

Reply:
`UpdateSolutionNativeMobileFirstValidator` doesn't implement the interface `IValidator<T>` so that needs to be impelemented. Here are my changes:

```c#
using NHSD.BuyingCatalogue.Solutions.Application.Commands.Validation;

namespace NHSD.BuyingCatalogue.Solutions.Application.Commands.ClientApplications.NativeMobile.UpdateSolutionNativeMobileFirst
{
    internal sealed class UpdateSolutionNativeMobileFirstValidator : IValidator<UpdateSolutionNativeMobileFirstCommand, ISimpleResult>
    {
        public ISimpleResult Validate(UpdateSolutionNativeMobileFirstCommand command)
            => new RequiredValidator()
                .Validate(command.MobileFirstDesign, "mobile-first-design")
                .Result();
    }
}
```
`UpdateSolutionNativeMobileFirstHandler.cs`:
```c#
public async Task<ISimpleResult> Handle(
    UpdateSolutionNativeMobileFirstCommand request,
    CancellationToken cancellationToken)
{
    var validationResult = _updateSolutionNativeMobileFirstValidator.Validate(request);

    if (validationResult.IsValid)
    {
        await _clientApplicationPartialUpdater.UpdateAsync(
            request.SolutionId,
            clientApplication =>
            {
                clientApplication.NativeMobileFirstDesign = request.MobileFirstDesign.ToBoolean();
            },
            cancellationToken);
    }

    return validationResult;
}
```

Reply:
Just merged those changes in 👍